services/sources/hathi_trust_service.py

```diff
@@ -0,0 +1,127 @@
+import csv
+from datetime import datetime
+import gzip
+from io import BytesIO
+import requests
+from typing import Optional, Generator
+
+from constants.get_constants import get_constants
+from mappings.hathitrust import HathiMapping
+from model import Record
+from .source_service import SourceService
+from logger import create_log
+
+
+logger = create_log(__name__)
+
+
+class HathiTrustService(SourceService):
+    HATHI_DATAFILES = 'https://www.hathitrust.org/files/hathifiles/hathi_file_list.json'
+    HATHI_RIGHTS_SKIPS = ['ic', 'icus', 'ic-world', 'und']
+    FIELD_SIZE_LIMIT = 131072 * 2 # 131072 is the default size limit
+
+    def __init__(self):
+        self.constants = get_constants()
+        self.record_count = 0
+
+    def get_records(
+        self, 
+        full_import: bool = False, 
+        start_timestamp: Optional[datetime] = None, 
+        offset: int = 0, 
+        limit: Optional[int] = None
+    ) -> Generator[Record, None, None]:
```

Comment:
It *might* be worth rethinking this method with a while loop or something instead of recursion - I found tracing how the record count var is used and set a little difficult to trace, especially, since it's an instance variable rather than just a simple param.  But that's more of a subjective thing, and I know unwinding this would be a bit of elbow grease!